docs/rfcs/semconv-feature-gates.md

```diff
@@ -0,0 +1,196 @@
+> [v1.36.0 of this document](https://github.com/open-telemetry/semantic-conventions/blob/v1.36.0/docs/gen-ai/README.md)
+> (or prior):
+>
+> * SHOULD NOT change the version of the GenAI conventions that they emit by default.
+>   Conventions include, but are not limited to, attributes, metric, span and event names,
+>   span kind and unit of measure.
+> * SHOULD introduce an environment variable `OTEL_SEMCONV_STABILITY_OPT_IN`
+>   as a comma-separated list of category-specific values. The list of values
+>   includes:
+>   * `gen_ai_latest_experimental` - emit the latest experimental version of
+>     GenAI conventions (supported by the instrumentation) and do not emit the
+>     old one (v1.36.0 or prior).
+>   * The default behavior is to continue emitting whatever version of the GenAI
+>     conventions the instrumentation was emitting (1.36.0 or prior).
+>
+> This transition plan will be updated to include stable version before the
+> GenAI conventions are marked as stable.
+
+</details>
+
+## Proposed mechanism
+
+Support the `<id>` (e.g. `hostmetrics`) `kind` (e.g. `receiver`) component is migrating from old to new
+semantic conventions on the area `area` (e.g. `process`). To support this, it defines two semantic
+conventions: `<kind>.<id>.EmitNew<Area>Conventions` (e.g.
+`receiver.hostmetrics.EmitNewProcessConventions`) and `<kind>.<id>.DontEmitOld<Area>Conventions`
+(e.g. `receiver.hostmetrics.DontEmitOldProcessConventions`). These feature gates work as follows:
+
+| `<kind>.<id>.EmitNew<Area>Conventions` status | `<kind>.<id>.DontEmitOld<Area>Conventions` status    | Resulting behavior                                        |
+|-----------------------------------------------|-------------------------------------------------------|-----------------------------------------------------------|
+| Disabled                                      | Disabled                                              | Emit telemetry under the 'old' conventions                |
+| Disabled                                      | Enabled                                               | Error at startup since this would not emit any telemetry  |
+| Enabled                                       | Disabled                                              | Emit telemetry under both the old and the new conventions |
+| Enabled                                       | Enabled                                               | Emit telemetry under the new conventions                  |
+
+Both feature gates evolve at the same pace through the feature gate stages, so that the progression
+is as follows:
+1. Initially both are at **alpha** stage (disabled by default). This means that the default behavior
+   is to emit only the 'old' conventions. Users can opt-in to emit the new conventions alongside the
+   old conventions or to emit only the new conventions.
+2. Whenever the semantic conventions are mature enough, the feature gates are promoted to the
```

Comment:
Continuing from https://github.com/open-telemetry/opentelemetry-collector/pull/14273/files#r2606069107, shall we make the `Whenever` more specific? From my perspective we should move to beta only when the semantic conventions are stable. So that should be the main checkpoint.

The reason is that beta feature gates might enforce users to switch and that should only happen once the semantic conventions are stable to avoid back and forth (i.e. semantic convention change from rc to development again for some reason).